File: Java Assignment/Bike.java
public class Bike{

	private boolean isStarted;
	private int speed;
	
	public void onBike(){
		isStarted = true;

	}
	
	public void OffBike(){
		if (isStarted == false) throw new IllegalArgumentException("You cannot stop bike when it is not started");
		isStarted = false;
	}
	
	public boolean getIsStarted(){
		return isStarted;

	}

	public void accelerateBike(){
		if (isStarted == false) throw new IllegalArgumentException("You cannot accelerate when the bike is off");
		else if (speed >= 40) setAccelerateToGearFour();
		else if (speed >= 30) setAccelerateToGearThree();
		else if (speed >= 20) setAccelerateToGearTwo();
		else setAccelerateToGearOne();

	}
	
	public void decelerateBike(){
		if (isStarted == false) throw new IllegalArgumentException("You cannot accelerate when the bike is off");
		else if (speed >= 40) setDecelerateToGearFour();
		else if (speed >= 30) setDecelerateToGearThree();
		else if (speed >= 20) setDecelerateToGearTwo();
		else if(speed >= 1) setDecelerateToGearOne();

	}

	private void setAccelerateToGearOne(){
		speed++;
	}

	
	private void setAccelerateToGearTwo(){
		speed += 2;							
	}

	
	private void setAccelerateToGearThree(){
		speed += 3;			
				
					

	}
	
	private void setAccelerateToGearFour(){
		speed += 4;
						

	}

	private void setDecelerateToGearOne(){
		speed--;
	}

	
	private void setDecelerateToGearTwo(){
		speed -= 2;							
	}

	
	private void setDecelerateToGearThree(){
		speed -= 3;			
	}
	
	private void setDecelerateToGearFour(){
		speed -= 4;
						

	}








	
	public int getSpeed(){
		return speed;


	}



}

File: Python/bike.py
class Bike:
    def __init__(self):
        self.speed = 0
        self.isStarted = False

    def on_bike(self):
            self.isStarted = True

    def off_bike(self):
        if self.isStarted == False:
            raise ValueError("Bike is not On")
        else:
            self.isStarted = False

    def get_bike_status(self):
        return self.isStarted

    def set_accelerate_to_gear_one(self):
        self.speed += 1


    def set_accelerate_to_gear_two(self):
        self.speed += 2

    def set_accelerate_to_gear_three(self):
        self.speed += 3

    def set_accelerate_to_gear_four(self):
        self.speed += 4

    def get_speed(self):
        return self.speed

    def accelerateBike(self):
        if self.isStarted == False:
            raise ValueError("Bike is not On")
        elif self.speed >= 40:
            self.set_accelerate_to_gear_four()
        elif self.speed >= 30:
            self.set_accelerate_to_gear_three()
        elif self.speed >= 20:
            self.set_accelerate_to_gear_two()
        else:
            self.set_accelerate_to_gear_one()
    def set_decelerate_to_gear_one(self):
        self.speed -= 1


    def set_decelerate_to_gear_two(self):
        self.speed -= 2

    def set_decelerate_to_gear_three(self):
        self.speed -= 3

    def set_decelerate_to_gear_four(self):
        self.speed -= 4


    def decelerateBike(self):
        if self.isStarted == False:
            raise ValueError("Bike is not On")
        elif self.speed >= 40:
            self.set_decelerate_to_gear_four()
        elif self.speed >= 30:
            self.set_decelerate_to_gear_three()
        elif self.speed >= 20:
            self.set_decelerate_to_gear_two()
        elif self.speed >= 1:
            self.set_decelerate_to_gear_one()